### DISCUSSION TITLE
Get WinGet Results in PowerShell

### DISCUSSION THREAD

**Discussion by OLLI-S** [microsoft/winget-pkgs#58046] [Q&A] [answered]
Hello,

For some noob family members (that always ask for a remote-session, so I can update their systems) I created a PowerShell script, that checks for available app-updates using WinGet:
```winget upgrade```

Is there a way to get a result from WinGet, that I can work with?
Like the number of available updates?

If there is such an result, I can check if the value is > 0 and in this case ask the user, if these updates should be automatically installed.
If the user selects YES, then I update all apps:
```winget upgrade --all```

Thank you for your help

OLLI

**Comment by denelon** [comment, marked as answer]
@OLLI-S 

We're working on a native PowerShell implementation:
* https://github.com/microsoft/winget-cli/issues/221

You should be able to safely run `winget upgrade --all`. The primary issue would be if any of the packages require UAC to continue. With Windows Package Manager 1.3 the "Unknown" versions are hidden to avoid the install loop.

**Reply by OLLI-S** [reply]
> With Windows Package Manager 1.3 the "Unknown" versions are hidden to avoid the install loop.

@denelon 
This will really help a lot, at my PC there are 2 updates shown due to the "Unknown" installed versions.
